### PR TITLE
Example/regression

### PR DESCRIPTION
**Pull request by ashdtu**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

#1148 

### Changes

Create an example to demonstrate how to do Regression problem using Burn framework. The example uses Toy-diabetes dataset present as a standard dataset in sklearn [datasets](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_diabetes.html). The dataset is also available for download from Hugging Face hub [dataset](https://huggingface.co/datasets/Jayabalambika/toy-diabetes). 

### Testing
* The dataset was split into train/test split and the MSE Loss for both splits were logged. No overfitting in the given model.
* ./run-checks.sh all was executed to check for compilation validity in example. No separate unit tests written.

### CODE REVIEW

File: examples/simple-regression/src/dataset.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        Self::new("train")
+    }
+
+    pub fn test() -> Self {
+        Self::new("test")
+    }
+
+    // pub fn get_full_data() -> SqliteDataset<DiabetesItem> {
+    //
+    // }
```
Comment (nathanielsimard) [inline examples/simple-regression/src/dataset.rs:85]: Dead code => should be removed.

File: examples/simple-regression/src/dataset.rs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+impl<B: Backend> DiabetesBatcher<B> {
+    pub fn new(device: B::Device) -> Self {
+        Self { device }
+    }
+}
+
+impl<B: Backend> Batcher<DiabetesItem, DiabetesBatch<B>> for DiabetesBatcher<B> {
+    fn batch(&self, items: Vec<DiabetesItem>) -> DiabetesBatch<B> {
+
```
Comment (nathanielsimard) [inline examples/simple-regression/src/dataset.rs:133]: Just a preference to remove empty lines at the beginning of functions :).

File: examples/simple-regression/src/dataset.rs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+        let targets = items
+            .iter()
+            .map(|item| Tensor::<B, 1>::from_floats([(item.response as f32)], &self.device))
+            .collect();
+
+        let targets  = Tensor::cat(targets, 0);
+        let targets = (targets.clone() - targets.clone().min_dim(0))
+            .div(targets.clone().max_dim(0) - targets.clone().min_dim(0));
```
Comment (nathanielsimard) [inline examples/simple-regression/src/dataset.rs:169]: Same

File: examples/simple-regression/src/model.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+};
+use burn::nn::loss::Reduction::Mean;
+use burn::nn::ReLU;
+
+#[derive(Module, Debug)]
+pub struct LinearModel<B: Backend> {
+    input_layer: Linear<B>,
+    output_layer: Linear<B>,
+    activation: ReLU
+}
```
Comment (nathanielsimard) [inline examples/simple-regression/src/model.rs:19]: I would rename it to `RegressionModel` since it's not technically linear (there is one activation function used).

File: examples/simple-regression/src/model.rs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            .with_bias(true).init(device);
+        let output_layer = LinearConfig::new(64, 1)
+            .with_bias(true).init(device);
+
+        Self {
+            input_layer,
+            output_layer,
+            activation: ReLU::new()
+        }
+    }
```
Comment (nathanielsimard) [inline examples/simple-regression/src/model.rs]: Instead of a new function, I would create a config:

```rust
#[derive(Config)]
pub struct RegressionModelConfig {
    pub num_features: usize;
    #[config(default = 64)]
    pub hidden_size: usize;
}
```
Then I would implement init

```rust
impl RegressionModelConfig {
    pub fn init<B: Backend>(&self, device: &B::Device) -> RegressionModel<B> {
            let input_layer = LinearConfig::new(self.num_features, self.hidden_size)
            .with_bias(true).init(device);
        let output_layer = LinearConfig::new(self.hidden_size, 1)
            .with_bias(true).init(device);

        Self {
            input_layer,
            output_layer,
            activation: ReLU::new()
        }
    }
}
```

Something like that.

File: examples/simple-regression/src/training.rs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    #[config(default = 2)]
+    pub num_workers: usize,
+
+    #[config(default = 42)]
+    pub seed: u64,
+
+    pub optimizer: SgdConfig,
+
+    #[config(default = 10)]
+    pub input_feature_len: usize,
```
Comment (nathanielsimard) [inline examples/simple-regression/src/training.rs:35]: This should be replaced by the model config.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
I've got some minor comments, but good job 👏

**Comment by ashdtu** [comment]
Hi, made all changes suggested in latest commit.

**Review by nathanielsimard — APPROVED** [review]
Very clean, good job! 👏
